nixos/modules/security/systemd-confinement.nix

```diff
@@ -183,6 +182,14 @@ in {
         echo "BindReadOnlyPaths=$realprog:/bin/sh" >> "$serviceFile"
       ''}
 
+      # If DynamicUser= is enabled, PrivateTmp= is implied (and cannot be turned off).
+      # but with "chroot-only" the tmp directories are not expected,
+      # So disable them unless PrivateTmp=true is explicitely set **in the NixOS configuration**.
+      ${lib.optionalString (cfg.confinement.mode == "chroot-only" && !cfg.serviceConfig.PrivateTmp) ''
```

Comment:
This also should not depend on `chroot-only`, even though it wouldn't increase the attack surface on `full-apivfs`.

Reply:
I've updated accordingly.